test/behaviors/integrationVault.ts

```diff
@@ -0,0 +1,799 @@
+import { expect } from "chai";
+import { BigNumber, Contract } from "ethers";
+import { TestContext } from "../library/setup";
+import Exceptions from "../library/Exceptions";
+import {
+    encodeToBytes,
+    mintUniV3Position_USDC_WETH,
+    randomAddress,
+    sleep,
+    withSigner,
+} from "../library/Helpers";
+import {
+    INTEGRATION_VAULT_INTERFACE_ID,
+    VAULT_REGISTRY_INTERFACE_ID,
+} from "../library/Constants";
+import { ERC20RootVault, ERC20Vault } from "../types";
+import { ethers } from "hardhat";
+import { randomHash } from "hardhat/internal/hardhat-network/provider/fork/random";
+import { PermissionIdsLibrary } from "../../deploy/0000_utils";
+
+export type IntegrationVaultContext<S extends Contract, F> = TestContext<
+    S,
+    F
+> & {
+    erc20Vault: ERC20Vault;
+    erc20RootVault: ERC20RootVault;
+    curveRouter: string;
+};
+
+export function integrationVaultBehavior<S extends Contract>(
+    this: IntegrationVaultContext<S, {}>,
+    {}: {}
+) {
+    const APPROVE_SELECTOR = "0x095ea7b3";
+    describe("#push", () => {
+        it("emits Push event", async () => {
+            await expect(
+                this.subject.push([this.usdc.address], [BigNumber.from(1)], [])
+            ).to.emit(this.subject, "Push");
+        });
+        it("passes when tokens transferred", async () => {
```

Comment:
It's better to name happy case to what it does exactly. E.g. `pushes tokens to the underlying protocol`